plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/cache/IcebergCacheKeyProvider.java

```diff
@@ -19,13 +19,20 @@
 import java.io.IOException;
 import java.util.Optional;
 
+import static java.util.Locale.ENGLISH;
+
 public class IcebergCacheKeyProvider
         implements CacheKeyProvider
 {
     @Override
     public Optional<String> getCacheKey(TrinoInputFile delegate)
             throws IOException
     {
-        return Optional.of(delegate.location().path());
+        String pathLowerCase = delegate.location().path().toLowerCase(ENGLISH);
+        // Iceberg data and metadata files are immutable
+        if (pathLowerCase.endsWith(".avro") || pathLowerCase.endsWith(".parquet") || pathLowerCase.endsWith(".orc") || pathLowerCase.endsWith(".json")) {
```

Comment:
Are there any other files apart from this it tries to cache ? If not we can keep it the earlier way it self.

Reply:
There are test related files from FileHiveMetastore which end up here and those files are mutable, I didn't find a cleaner to solution to handling that.

Reply:
since these are immutable, why we need extra if (their modification time shouldn't change)?

Reply:
As mentioned above, we hit this code path for files used by FileHiveMetastore in tests as well and those files happen to be mutable. Handling it explicitly here was the least disruptive way I could see

Reply:
> If we restrict caching without metadata operations to only these extensions we may be unable to cache data in Iceberg tables written by some engines, or some types of migrated Iceberg tables. (for instance, Delta tables written by Trino don't use an extension on their data files. If these tables were migrated to Iceberg we would not be able to cache them).

When the file extension doesn't match, we still cache the file, but with modification time key. This will not result in extra file status call for data files on worker as the file status is already available from ConnectorSplit there.
I assume that file extension will always match for metadata files.

Reply:
Even I don't know any better solution for avoiding .trinopermission or .trinoschema files. so this should be fine.

Reply:
I've simplified this now to just explicitly exclude FileHiveMetastore files